client/post-editor/media-modal/detail/detail-preview-videopress.jsx

```diff
@@ -1,47 +1,102 @@
 /**
  * External dependencies
  */
-const React = require( 'react' );
+import React, { Component, PropTypes } from 'react';
+import debug from 'debug';
 
 /**
  * Internal dependencies
  */
-const loadScript = require( 'lib/load-script' ).loadScript;
+import loadScript from 'lib/load-script';
 
-module.exports = React.createClass( {
-	displayName: 'EditorMediaModalDetailPreviewVideoPress',
+/**
+ * Module variables
+ */
+const log = debug( 'calypso:post-editor:videopress' );
+const videoPressUrl = 'https://wordpress.com/wp-content/plugins/video/assets/js/next/videopress.js';
+
+class EditorMediaModalDetailPreviewVideoPress extends Component {
+	static propTypes = {
+		isPlaying: PropTypes.bool,
+		item: PropTypes.object.isRequired,
+	};
+
+	static defaultProps = {
+		isPlaying: false,
+	};
 
-	propTypes: {
-		item: React.PropTypes.object.isRequired
-	},
+	constructor() {
+		super();
+
+		this.onScriptLoaded = this.onScriptLoaded.bind( this );
+	}
 
 	componentDidMount() {
 		this.loadInitializeScript();
-	},
+	}
+
+	componentWillUnmount() {
+		loadScript.removeScriptCallback( videoPressUrl, this.onScriptLoaded );
+		this.destroy();
+	}
 
 	componentDidUpdate( prevProps ) {
 		if ( this.props.item.videopress_guid !== prevProps.item.videopress_guid ) {
+			this.destroy();
 			this.loadInitializeScript();
 		}
-	},
+	}
+
+	shouldComponentUpdate( nextProps ) {
+		if ( this.props.item.videopress_guid !== nextProps.item.videopress_guid ) {
+			return true;
+		}
+
+		return false;
+	}
+
+	setVideoInstance = ref => this.video = ref;
 
 	loadInitializeScript() {
-		loadScript( 'https://videopress.com/videopress-iframe.js' );
-	},
+		loadScript.loadScript( videoPressUrl, this.onScriptLoaded );
+	}
 
-	getEmbedUrl() {
-		return `https://videopress.com/embed/${ this.props.item.videopress_guid }`;
-	},
+	onScriptLoaded( error ) {
+		if ( error ) {
+			log( `Script${ error.src } failed to load.` );
+			return;
+		}
+
+		/* eslint-disable no-undef */
+		this.player = videopress( this.props.item.videopress_guid, this.video, {
```

Comment:
To avoid `no-undef` disabling, might consider calling on `window.videopress` or on the `global` object, optionally with a check to first ensure that it exists (in case future versions of the script change and no longer set the assumed global).